File: stater/src/test/java/ru/alexpanchenko/stater/plugin/StateFieldStorageTest.java
package ru.alexpanchenko.stater.plugin;

import org.junit.Test;

import ru.alexpanchenko.stater.plugin.model.SaverField;
import ru.alexpanchenko.stater.plugin.model.StateType;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class StateFieldStorageTest {

  private final SaverField field = new SaverField("n", "d", "s", "o", StateType.BOOLEAN);

  @Test
  public void testAddField() {
    final StateFieldStorage fieldStorage = new StateFieldStorage();
    fieldStorage.add(field);
    assertEquals(fieldStorage.getFields().size(), 1);
    assertEquals(fieldStorage.getFields().get(0), field);
  }

  @Test
  public void testClearField() {
    final StateFieldStorage fieldStorage = new StateFieldStorage();
    fieldStorage.add(field);
    fieldStorage.add(field);
    assertEquals(fieldStorage.getFields().size(), 2);
    fieldStorage.clear();
    assertTrue(fieldStorage.getFields().isEmpty());
  }
}

File: stater/src/test/kotlin/ru/alexpanchenko/stater/plugin/StateFieldStorageTest.kt
package ru.alexpanchenko.stater.plugin

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test
import ru.alexpanchenko.stater.plugin.model.SaverField
import ru.alexpanchenko.stater.plugin.model.StateType

class StateFieldStorageTest {

  private val field = SaverField("n", "d", "s", "o", StateType.BOOLEAN)

  @Test
  fun testAddField() {
    val fieldStorage = StateFieldStorage()
    fieldStorage.add(field)
    assertEquals(fieldStorage.fields.size.toLong(), 1)
    assertEquals(fieldStorage.fields[0], field)
  }

  @Test
  fun testClearField() {
    val fieldStorage = StateFieldStorage()
    fieldStorage.add(field)
    fieldStorage.add(field)
    assertEquals(fieldStorage.fields.size.toLong(), 2)
    fieldStorage.clear()
    assertTrue(fieldStorage.fields.isEmpty())
  }
}